src/com/trilead/ssh2/crypto/dh/EcDhExchange.java

```diff
@@ -0,0 +1,133 @@
+package com.trilead.ssh2.crypto.dh;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECKey;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.ECPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+
+import javax.crypto.KeyAgreement;
+import javax.management.RuntimeErrorException;
+
+import com.trilead.ssh2.signature.ECDSAKeyAlgorithm;
+
+
+
+/**
+ * @author kenny
+ *
+ */
+public class EcDhExchange extends GenericDhExchange {
+	private ECPrivateKey clientPrivate;
+	private ECPublicKey clientPublic;
+	private ECPublicKey serverPublic;
+	private ECDSAKeyAlgorithm ecdsKeyAlgorithm;
+
+	@Override
+	public void init(String name) throws IOException {
+		final ECParameterSpec spec;
+
+		if ("ecdh-sha2-nistp256".equals(name)) {
```

Comment:
Same nitpick here; this could use a switch statement.

Reply:
Done!